thunder/tests/test_jit_general.py

```diff
@@ -1200,6 +1200,45 @@ def forward(self, x):
     gradcheck(jitted, (x,))
 
 
+def test_autograd_function_apply():
+
+    def forward(ctx, x):
+        saved_for_backward = (x,)
+        return x.sin(), saved_for_backward
+
+    def backward(ctx, grad_output, *saved_tensors):
+        (x,) = saved_tensors
+        return grad_output * x.cos()
+
+    def my_sin(x):
+        return torch.ops.higher_order.autograd_function_apply(
+            forward,
+            backward,
+            x,
+            args_tensor_mask=[True],
+        )
+
+    jitted = thunder.jit(my_sin)
```

Comment:
`thunder.last_traces(jitted)[0]` is as follows:

```python
import thunder.torch as ltorch
import torch
from thunder.executors.torchex import no_autocast

@torch.no_grad()
@no_autocast
def computation(x):
  # x: "cpu f32[2, 2]"
  # output of fwd_body: <TensorProxy(name="t0", dtype=thunder.dtypes.float32, shape=(2, 2))>, saved_values from fwd_body: ['t_0']
  t0 = ltorch.autograd_function_apply_cegfp(x)  # t0: "cpu f32[2, 2]"
    # t0 = ltorch.sin(x)  # t0: "cpu f32[2, 2]"
      # t0 = prims.sin(x)  # t0: "cpu f32[2, 2]"

  # /path/to/lightning-thunder/thunder/tests/test_jit_general.py:1211:                 return grad_output * x.cos()
  return t0
```

Reply:
Cool! In the lowering passes this `autograd_function_apply_cegfp` is decomposed using subsymbols because no pass has a rule to handle `autograd_function_apply_cegfp` as a composite, right?

The backward definition is currently ignored. I think it would be better if the backward function in this test raises an error with a comment to be clear about it.